Calibration/EcalAlCaRecoProducers/python/ALCARECOEcalUncalIsolElectron_cff.py

```diff
@@ -26,7 +26,7 @@
 from RecoLocalCalo.EcalRecProducers.ecalGlobalUncalibRecHit_cfi import *
 ecalUncalibRecHitSequence53X = cms.Sequence(ecalGlobalUncalibRecHit * ecalDetIdToBeRecovered)
 
-uncalibRecHitSeq = cms.Sequence( (ecalDigis + ecalPreshowerDigis) * ecalUncalibRecHitSequence)
+uncalibRecHitSeq = cms.Sequence(ecalUncalibRecHitSequence, cms.Task(ecalDigis, ecalPreshowerDigis))
```

Comment:
@Dr15Jones is there some documentation to understand the reasoning behind this change?

Reply:
Is the commit message sufficient?
https://github.com/cms-sw/cmssw/pull/20502/commits/8fec79ea4c53489e4dce02e75d839007185ffa46

Reply:
I guess my problem is that I am a bit lacking understanding of the `cms.Task` concept.
What extra information does it provide to wrap `ecalDigis + ecalPreshowerDigis` in a `cms.Task`.
Sorry for my previous ignorance...

Reply:
You can find the full documentation about Tasks here
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideAboutPythonConfigFile#Task_Objects

Essentially, modules on a Task will be run when their data products are first needed by another module (it is what we used to call 'unscheduled'). In contrast, a Sequence runs its modules in a specific order. However, in this case, the Sequence order was causing an circular dependency between the modules. Moving the modules to a Task, however, allows the framework to run the modules only when needed and this avoids the circle (since the order the modules run is strictly based on data dependency and not by an external ordering).

Reply:
After making a sketch of your commit message it was literally a circular dependency.
Thanks for the thorough explanation!